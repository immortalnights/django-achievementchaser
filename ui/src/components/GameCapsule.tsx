import { Typography, Tooltip } from "@mui/material"
import dayjs from "dayjs"
import { useMemo } from "react"
import { getRelativeTime, formatDate } from "../dayjsUtilities"
import Link from "./Link"
import GameIcon from "./GameIcon"

const GameCapsule = ({
    player,
    game,
    ownedGame,
}: {
    player: string
    game: Game
    ownedGame?: Omit<PlayerOwnedGame, "game">
}) => {
    let completionTitle: string | undefined

    if (ownedGame && ownedGame.completed) {
        const completionDate = dayjs(ownedGame.completed)

        completionTitle = `Completed: ${completionDate.format(
            "MMM D, YYYY"
        )} (${getRelativeTime(completionDate)})`
    } else if (
        game.achievementCount &&
        ownedGame &&
        ownedGame.unlockedAchievementCount
    ) {
        const percentage =
            (ownedGame.unlockedAchievementCount / game.achievementCount) * 100

        completionTitle = `Progress: ${ownedGame.unlockedAchievementCount} / ${
            game.achievementCount
        } (${percentage.toFixed(2)}%)`
    } else if (game.achievementCount) {
        completionTitle = `Achievements: ${game.achievementCount}`
    }

    let difficultyPercentageTitle: string | undefined
    if (game.difficultyPercentage) {
        difficultyPercentageTitle = `Difficulty: ${game.difficultyPercentage.toFixed(
            2
        )}%`
    }

    let lastPlayedTitle: string | undefined
    if (ownedGame && ownedGame.lastPlayed) {
        lastPlayedTitle = `Last Played: ${formatDate(ownedGame.lastPlayed)}`
    }

    const titleEl = useMemo(
        () => (
            <>
                <Typography>{game.name}</Typography>
                {completionTitle && (
                    <Typography fontSize="small">{completionTitle}</Typography>
                )}
                {lastPlayedTitle && (
                    <Typography fontSize="small">{lastPlayedTitle}</Typography>
                )}
                {difficultyPercentageTitle && (
                    <Typography fontSize="small">
                        {difficultyPercentageTitle}
                    </Typography>
                )}
            </>
        ),
        [game.name, completionTitle, lastPlayedTitle, difficultyPercentageTitle]
    )

    return (
        <Link to={`/Player/${player}/Game/${game.id}`}>
            <Tooltip title={titleEl} arrow enterDelay={500} leaveDelay={0}>
                <GameIcon {...game} />
            </Tooltip>
        </Link>
    )
}

export default GameCapsule
