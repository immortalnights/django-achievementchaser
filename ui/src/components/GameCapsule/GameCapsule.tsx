import { Tooltip, Typography } from "@mui/material"
import dayjs from "dayjs"
import { ReactNode, forwardRef } from "react"
import { getRelativeTime, formatDate } from "@/dayjsUtilities"
import Link from "../Link"
import GameIcon from "../GameIcon"

export const GameIconWithTooltip = ({
    game,
    tooltip,
}: {
    game: Game
    tooltip: ReactNode
}) => {
    const Wrapped = forwardRef((props, ref) => (
        <GameIcon {...game} {...props} innerRef={ref} />
    ))

    return (
        <Tooltip title={tooltip} arrow enterDelay={500} leaveDelay={0}>
            <Wrapped />
        </Tooltip>
    )
}

const TooltipContent = ({
    game,
    ownedGame,
}: {
    game: Game
    ownedGame?: Omit<PlayerOwnedGame, "game">
}) => {
    let completionTitle: string | undefined
    let lastPlayedTitle: string | undefined

    if (ownedGame) {
        if (ownedGame.completed) {
            const completionDate = dayjs(ownedGame.completed)

            completionTitle = `Completed: ${formatDate(
                completionDate
            )} (${getRelativeTime(completionDate)})`
        } else if (
            game.achievementCount &&
            ownedGame.unlockedAchievementCount
        ) {
            const percentage =
                (ownedGame.unlockedAchievementCount / game.achievementCount) *
                100

            completionTitle = `Progress: ${
                ownedGame.unlockedAchievementCount
            } of ${game.achievementCount} (${percentage.toFixed(2)}%)`
        }

        if (ownedGame.lastPlayed) {
            lastPlayedTitle = `Last Played: ${formatDate(
                ownedGame.lastPlayed
            )} (${getRelativeTime(ownedGame.lastPlayed)})`
        }
    } else if (game.achievementCount) {
        completionTitle = `Achievements: ${game.achievementCount}`
    }

    let difficultyPercentageTitle: string | undefined
    if (game.difficultyPercentage) {
        difficultyPercentageTitle = `Difficulty: ${game.difficultyPercentage.toFixed(
            2
        )}%`
    }

    return (
        <>
            <Typography>{game.name}</Typography>
            {completionTitle && (
                <Typography fontSize="small">{completionTitle}</Typography>
            )}
            {lastPlayedTitle && (
                <Typography fontSize="small">{lastPlayedTitle}</Typography>
            )}
            {difficultyPercentageTitle && (
                <Typography fontSize="small">
                    {difficultyPercentageTitle}
                </Typography>
            )}
        </>
    )
}

const GameCapsule = ({
    player,
    game,
    ownedGame,
}: {
    game: Game
    player?: string
    ownedGame?: Omit<PlayerOwnedGame, "game">
}) => {
    const linkTo = player
        ? `/Player/${player}/Game/${game.id}`
        : `/Game/${game.id}`

    return (
        <Link to={linkTo}>
            <GameIconWithTooltip
                game={game}
                tooltip={<TooltipContent game={game} ownedGame={ownedGame} />}
            />
        </Link>
    )
}

export default GameCapsule
