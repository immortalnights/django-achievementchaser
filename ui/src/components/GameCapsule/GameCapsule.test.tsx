import renderer from "react-test-renderer"
import { expect, test } from "vitest"
import GameCapsule from "."
import { MemoryRouter } from "react-router-dom"
import {
    neverPlayedOwnedGame,
    perfectPlayerOwnedGame,
    playedPlayerOwnedGame,
} from "@test/data"

test("snapshot", () => {
    const game = {
        name: "Crusader Kings III",
        id: "1158310",
        imgIconUrl: "8a0d88dfaff790ea1aa2b9fcf50d4e3b4f49cf56",
        achievementCount: 117,
        difficultyPercentage: 0.10000000149011612,
    } satisfies Game

    // game
    // eslint-disable-next-line
    const gameCapsule = renderer.create(
        <MemoryRouter>
            <GameCapsule game={game} />
        </MemoryRouter>
    )
    expect(gameCapsule.toJSON()).toMatchSnapshot()

    // game, link to player
    // eslint-disable-next-line
    const gameLinkToPlayer = renderer.create(
        <MemoryRouter>
            <GameCapsule game={game} player="1234" />
        </MemoryRouter>
    )
    expect(gameLinkToPlayer.toJSON()).toMatchSnapshot()

    // game with owned game information
    // eslint-disable-next-line
    const neverPlayedGame = renderer.create(
        <MemoryRouter>
            <GameCapsule game={game} ownedGame={neverPlayedOwnedGame} />
        </MemoryRouter>
    )
    expect(neverPlayedGame.toJSON()).toMatchSnapshot()

    // eslint-disable-next-line
    const perfectGame = renderer.create(
        <MemoryRouter>
            <GameCapsule game={game} ownedGame={perfectPlayerOwnedGame} />
        </MemoryRouter>
    )
    expect(perfectGame.toJSON()).toMatchSnapshot()

    // game with owned game information
    // eslint-disable-next-line
    const playedGame = renderer.create(
        <MemoryRouter>
            <GameCapsule game={game} ownedGame={playedPlayerOwnedGame} />
        </MemoryRouter>
    )
    expect(playedGame.toJSON()).toMatchSnapshot()
})
