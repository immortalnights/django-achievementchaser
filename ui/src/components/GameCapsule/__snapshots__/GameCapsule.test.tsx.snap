// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1qdg4h9-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1qdg4h9-MuiLink-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1pbzsj4-MuiTypography-root-MuiLink-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:2.125rem;line-height:1.235;letter-spacing:0.00735em;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1pbzsj4-MuiTypography-root-MuiLink-root"
  />,
  <a
    className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineNone css-1pbzsj4-MuiTypography-root-MuiLink-root"
    href="/Game/1158310"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      alt="Crusader Kings III"
      aria-label={null}
      aria-labelledby={null}
      className=" css-mt3xud"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      src="https://media.steampowered.com/steam/apps/1158310/capsule_184x69.jpg"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-pnutvq-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css pnutvq-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-viryjt-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css viryjt-MuiPopper-root-MuiTooltip-popper"
    />
  </a>,
]
`;

exports[`snapshot 2`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1qdg4h9-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1qdg4h9-MuiLink-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1pbzsj4-MuiTypography-root-MuiLink-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:2.125rem;line-height:1.235;letter-spacing:0.00735em;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1pbzsj4-MuiTypography-root-MuiLink-root"
  />,
  <a
    className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineNone css-1pbzsj4-MuiTypography-root-MuiLink-root"
    href="/Player/1234/Game/1158310"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      alt="Crusader Kings III"
      aria-label={null}
      aria-labelledby={null}
      className=" css-mt3xud"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      src="https://media.steampowered.com/steam/apps/1158310/capsule_184x69.jpg"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-pnutvq-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css pnutvq-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-viryjt-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css viryjt-MuiPopper-root-MuiTooltip-popper"
    />
  </a>,
]
`;

exports[`snapshot 3`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1qdg4h9-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1qdg4h9-MuiLink-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1pbzsj4-MuiTypography-root-MuiLink-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:2.125rem;line-height:1.235;letter-spacing:0.00735em;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1pbzsj4-MuiTypography-root-MuiLink-root"
  />,
  <a
    className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineNone css-1pbzsj4-MuiTypography-root-MuiLink-root"
    href="/Game/1158310"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      alt="Crusader Kings III"
      aria-label={null}
      aria-labelledby={null}
      className=" css-mt3xud"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      src="https://media.steampowered.com/steam/apps/1158310/capsule_184x69.jpg"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-pnutvq-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css pnutvq-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-viryjt-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css viryjt-MuiPopper-root-MuiTooltip-popper"
    />
  </a>,
]
`;

exports[`snapshot 4`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1qdg4h9-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1qdg4h9-MuiLink-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1pbzsj4-MuiTypography-root-MuiLink-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:2.125rem;line-height:1.235;letter-spacing:0.00735em;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1pbzsj4-MuiTypography-root-MuiLink-root"
  />,
  <a
    className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineNone css-1pbzsj4-MuiTypography-root-MuiLink-root"
    href="/Game/1158310"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      alt="Crusader Kings III"
      aria-label={null}
      aria-labelledby={null}
      className=" css-mt3xud"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      src="https://media.steampowered.com/steam/apps/1158310/capsule_184x69.jpg"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-pnutvq-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css pnutvq-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-viryjt-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css viryjt-MuiPopper-root-MuiTooltip-popper"
    />
  </a>,
]
`;

exports[`snapshot 5`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1qdg4h9-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1qdg4h9-MuiLink-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1pbzsj4-MuiTypography-root-MuiLink-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:2.125rem;line-height:1.235;letter-spacing:0.00735em;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;}",
      }
    }
    data-emotion="css 1pbzsj4-MuiTypography-root-MuiLink-root"
  />,
  <a
    className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineNone css-1pbzsj4-MuiTypography-root-MuiLink-root"
    href="/Game/1158310"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-mt3xud{box-shadow:0 8px 17px 2px rgba(0, 0, 0, 0.14),0 3px 14px 2px rgba(0, 0, 0, 0.12),0 5px 5px -3px rgba(0, 0, 0, 0.2);border:1px solid black;display:block;}",
        }
      }
      data-emotion="css mt3xud"
    />
    <img
      alt="Crusader Kings III"
      aria-label={null}
      aria-labelledby={null}
      className=" css-mt3xud"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      src="https://media.steampowered.com/steam/apps/1158310/capsule_184x69.jpg"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-pnutvq-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css pnutvq-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-viryjt-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"bottom\\"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"top\\"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"right\\"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*=\\"left\\"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css viryjt-MuiPopper-root-MuiTooltip-popper"
    />
  </a>,
]
`;
